packages/suite/src/components/wallet/Fees/components/CustomFee.tsx

```diff
@@ -87,26 +91,105 @@ const SetDefaultLimit = ({ onClick }: { onClick: () => void }) => (
 const FEE_PER_UNIT = 'feePerUnit';
 const FEE_LIMIT = 'feeLimit';
 
-const FeeCustom = ({
-    account: { networkType },
+type FormMethods = UseFormMethods<{
+    selectedFee?: FeeLevel['label'];
+    feePerUnit?: string;
+    feeLimit?: string;
+    estimatedFeeLimit?: string;
+}>;
+
+interface CustomFeeProps {
+    networkType: Account['networkType'];
+    feeInfo: FeeInfo;
+    errors: FormMethods['errors'];
+    register: (rules?: TypedValidationRules) => (ref: any) => void;
+    getValues: FormMethods['getValues'];
+    setValue: FormMethods['setValue'];
+    changeFeePerUnit?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+    changeFeeLimit?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+export const CustomFee = ({
+    networkType,
     feeInfo,
     errors,
     register,
     getValues,
     setValue,
     changeFeePerUnit,
     changeFeeLimit,
-}: Props) => {
+}: CustomFeeProps) => {
     const { maxFee, minFee } = feeInfo;
-    const feePerUnitValue = getValues(FEE_PER_UNIT);
-    const feePerUnitError = errors.feePerUnit;
 
-    const useFeeLimit = networkType === 'ethereum';
+    const feePerUnitValue = getValues(FEE_PER_UNIT);
     const feeLimitValue = getValues(FEE_LIMIT);
     const estimatedFeeLimit = getValues('estimatedFeeLimit') || ETH_DEFAULT_GAS_LIMIT;
+
+    const feePerUnitError = errors.feePerUnit;
     const feeLimitError = errors.feeLimit;
+
+    const useFeeLimit = networkType === 'ethereum';
     const feeLimitDisabled = false;
 
+    const validateFeeLimit = (value: string) => {
+        const feeBig = new BigNumber(value);
+
+        if (feeBig.isNaN()) {
+            return 'CUSTOM_FEE_IS_NOT_NUMBER';
+        }
+
+        // allow decimals in ETH since GWEI is not a satoshi
+        if (networkType !== 'ethereum' && networkType !== 'bitcoin' && !isInteger(value)) {
+            return 'CUSTOM_FEE_IS_NOT_INTEGER';
+        }
+
+        if (feeBig.lt(estimatedFeeLimit)) {
+            return (
+                <SetDefaultLimit
+                    onClick={() => {
+                        setValue(FEE_LIMIT, estimatedFeeLimit, {
+                            shouldValidate: true,
+                        });
+                    }}
+                />
+            );
+        }
+    };
+
+    const validateFee = (value: string) => {
+        const feeBig = new BigNumber(value);
+
+        if (feeBig.isNaN()) {
+            return 'CUSTOM_FEE_IS_NOT_NUMBER';
+        }
+
+        // allow decimals in ETH since GWEI is not a satoshi
+        if (networkType !== 'ethereum' && networkType !== 'bitcoin' && !isInteger(value)) {
+            return 'CUSTOM_FEE_IS_NOT_INTEGER';
+        }
+
+        // GWEI: 9 decimal places
+        if (networkType === 'ethereum' && !isDecimalsValid(value, 9)) {
```

Comment:
Applies for BTC too?

Reply:
Done but still confused if it should be 2 or 3.

Reply:
Better to ask @hynek-jina 

How many decimals do we want to allow for BTC fee?

Reply:
This is not related to mSats commonly used on Lightning Network. Here is the smallest unit of resulting fee 1 sat. For example if the transaction has 170 vB, then the 1.1 sat/vB will result in 187 sats. If you type 1.099999 it will end up with a same result of 187 sats fee.
So it is not necessary to introduce the limit on decimals.. But I would stick to two decimals limit.